frontend/src/js/components/modals/CachedFiles/Tabs/FileTab.jsx

```diff
@@ -244,34 +245,70 @@ class FileTab extends Component {
         }
     }
 
-    handleDragStart(event, file){
+    handleDragStart(event, file, option){
         let data = _.assign({}, file)
         event.dataTransfer.setData('text', JSON.stringify(data)); // datatype must be text/plain due to chrome bug
-        
-        this.setState({showBookmarkZone: true})
+        if(option == 'add'){
+            this.setState({showBookmarkZone: true})
+        }
+        else if(option == 'remove'){
+            this.setState({showRemoveBookmarkZone: true})
+        }
     }
 
     /* istanbul ignore next */
-    handleDragOver(event){
+    handleDragOver(event, option){
         event.preventDefault()
         event.stopPropagation() // Stupid drag and drop api issue
     }
 
-    handleDrop(event){
+    handleDrop(event, option){
         try {
             let file = JSON.parse(event.dataTransfer.getData('text'));
             let bookmarks = this.state.bookmarkFiles.slice()
-            bookmarks.push(file)
+            let fileInBookmarks = false
+            let fileInBookmarksIndex = -1
+            bookmarks.reduce((prev_val, current_obj) => {
```

Comment:
May need to check the logic around here. Adding bookmarks doesnt always work and removing often removes the wrong bookmark.

Reply:
:ballot_box_with_check: